raster/src/main/scala/geotrellis/raster/VectorToRaster.scala

```diff
@@ -129,13 +135,16 @@ object VectorToRaster {
                 var ws = 0.0
                 val length = pts.size
 
-                cfor(0)(_ < length, _ + 1) { i =>
+                var i = 0
+                while (i < length) {
                   val point = pts(i)
                   val dX = (destX - point.geom.x)
                   val dY = (destY - point.geom.y)
                   val d = dX * dX + dY * dY
+                  i += 1
+
                   if (d < rr) {
-                    val w = 1 / d
+                    val w = if (d == 0) w0 else 1 / d
```

Comment:
That's what I actually did on the first commit, but then I thought than
there are cases where you may have several point values for the same cell
and in that case we would be ignoring other points if there is one which
has coordinates equal to cell center...

In my opinion, which solution is better for you depends on what you are
trying to do with the interpolation.

For that reason I posted the two solutions in separate commits so we can
comment them.

El vie., 22 abr. 2016 17:03, Eugene Cheipesh <notifications@github.com>
escribió:

> In raster/src/main/scala/geotrellis/raster/VectorToRaster.scala
> <https://github.com/geotrellis/geotrellis/pull/1454#discussion_r60753341>:
>
> >                    if (d < rr) {
> > -                    val w = 1 / d
> > +                    val w = if (d == 0) w0 else 1 / d
>
> This fixes the NoData problem for when we have a point exactly for the
> co-ordinates we're trying to interpolate but not sure this is totally
> correct. Probably we should short-circuit the interpolation and return the
> actual value of the point. As d tends to 0 the weight tends to infinity, so
> it should dominate any other values. Also currently I think its possible
> for a point that is slightly off the focus co-ordinates to have more weight
> than the actual point.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/geotrellis/geotrellis/pull/1454/files/ac129192a182232c48d7a8d765c9deb25e64460d#r60753341>
>